src/cmp.rs

```diff
@@ -1,38 +1,94 @@
 use crate::iter::{iterate_lexical, iterate_lexical_only_alnum};
-use core::cmp::Ordering;
-
-macro_rules! cmp_ascii_digits {
-    (first_digits($lhs:ident, $rhs:ident), iterators($iter1:ident, $iter2:ident)) => {
-        let mut n1 = ascii_to_u64($lhs);
-        let mut n2 = ascii_to_u64($rhs);
-        loop {
-            match (
-                $iter1.peek().copied().filter(|c| c.is_ascii_digit()),
-                $iter2.peek().copied().filter(|c| c.is_ascii_digit()),
-            ) {
-                (Some(lhs), Some(rhs)) => {
-                    n1 = n1 * 10 + ascii_to_u64(lhs);
-                    n2 = n2 * 10 + ascii_to_u64(rhs);
-                    let _ = $iter1.next();
-                    let _ = $iter2.next();
-                }
-                (Some(_), None) => return Ordering::Greater,
-                (None, Some(_)) => return Ordering::Less,
-                (None, None) => {
-                    if n1 != n2 {
-                        return n1.cmp(&n2);
+use core::{
+    cmp::Ordering,
+    iter::Peekable,
+};
+
+fn cmp_ascii_digits(lhs: &mut Peekable<impl Iterator<Item=char>>, rhs: &mut Peekable<impl Iterator<Item=char>>) -> Option<Ordering> {
+    // The loop below iterates through both iterators at once and handles ascii digits for comparison.
+    // If one iterator runs out of ascii digits, it is stored in this enum, where each item,
+    // Lhs and Rhs, represents where the character originated from.
+    enum NonDigit {
```

Comment:
This may be a bit overengineered for what can also be tracked with a `(char, bool)`, but I think/hope this makes the use more clear.